hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/storefiletracker/StoreFileTrackerFactory.java

```diff
@@ -92,7 +92,7 @@ static String getStoreFileTrackerName(Class<? extends StoreFileTracker> clazz) {
     return name != null ? name.name() : clazz.getName();
   }
 
-  private static Class<? extends StoreFileTracker> getTrackerClass(Configuration conf) {
+  public static Class<? extends StoreFileTracker> getTrackerClass(Configuration conf) {
```

Comment:
So if we move the method to this class, then we do not need to change this modifier.

Reply:
CustomSFT param validation before snapshot cloning also relies on this.

Reply:
Seems OK to have these public here since `checkForRestoreSnapshot` method is also used in `SnapshotManager`. Seems overkill to add an interface to this Factory just for these two ("internal public api") methods, which is what we'd normally do. For a Factory class, I would expect I could get/create things as the caller.

Reply:
We need to load at least the implementation classes so we need to use some internal methods in StoreFileTrackerFactory. Anyway, not a blocker here, maybe we could introduce a special SFTChecker class, put it in the same package with SFTFactory, so we do not need to make these methods public while do not need to put the check methods in SFTFactory.